unit_test/live_model_test.cc

```diff
@@ -131,6 +131,56 @@ BOOST_AUTO_TEST_CASE(live_model_ranking_request) {
   BOOST_CHECK_EQUAL(status.get_error_msg(), "");
 }
 
+BOOST_AUTO_TEST_CASE(live_model_ranking_request_online_mode) {
+  //create a simple ds configuration
+  u::configuration config;
+  cfg::create_from_json(JSON_CFG, config);
+  config.set(r::name::EH_TEST, "true");
+  config.set(r::name::LEARNING_MODE, "ONLINE_MODE");
+
+  r::api_status status;
+
+  //create the ds live_model, and initialize it with the config
+  r::live_model ds = create_mock_live_model(config);
+  BOOST_CHECK_EQUAL(ds.init(&status), err::success);
+
+  const auto event_id = "event_id";
+  r::ranking_response response;
+  ds.choose_rank(event_id, JSON_CONTEXT, response, &status);
+  BOOST_CHECK_EQUAL(status.get_error_code(), 0);
+  BOOST_CHECK_EQUAL(status.get_error_msg(), "");
+}
+BOOST_AUTO_TEST_CASE(live_model_ranking_request_imitation_mode) {
+  //create a simple ds configuration
+  u::configuration config;
+  cfg::create_from_json(JSON_CFG, config);
+  config.set(r::name::EH_TEST, "true");
+  config.set(r::name::LEARNING_MODE, "IMITATION_MODE");
+
+  r::api_status status;
+
+  //create the ds live_model, and initialize it with the config
+  r::live_model ds = create_mock_live_model(config);
+  BOOST_CHECK_EQUAL(ds.init(&status), err::success);
+
+  const auto event_id = "event_id";
+
+  r::ranking_response response;
+
+  ds.choose_rank(event_id, JSON_CONTEXT, response, &status);
```

Comment:
>JSON_CONTEXT [](start = 27, length = 12)

JSON_CONTEXT has no actions, so are we testing anything here?

Reply:
Added a new JSON_CONTEXT_LEARNING and I verified online mode and imitation mode could give different suggestions.

Due to exploration, the selected action in online mode is not asserted.